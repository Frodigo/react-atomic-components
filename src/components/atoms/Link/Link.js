import React from 'react';
import PropTypes from 'prop-types';

import classes from '../Button/Button.module.css';

/**
 * This is the button.
 * @param {*} props
 */
const Link = (props) => {
  const { variant = 'primary', children, href, ...rest } = props;

  let buttonClass;

  switch (variant) {
    case 'primary': {
      buttonClass = classes.primary;

      break;
    }

    case 'secondary': {
      buttonClass = classes.secondary;

      break;
    }

    case 'tiertary': {
      buttonClass = classes.tiertary;

      break;
    }

    case 'inverse': {
      buttonClass = classes.inverse;

      break;
    }

    default: {
      buttonClass = null;
    }
  }

  return (
    <a href={href} className={buttonClass} {...rest}>
      {children}
    </a>
  );
};

Link.propTypes = {
  active: PropTypes.string,
  children: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  href: PropTypes.string.isRequired,
  variant: PropTypes.oneOf([
    'text',
    'primary',
    'secondary',
    'tiertary',
    'inverse',
  ]),
};

Link.defaultProps = {
  variant: 'text',
};

export default Link;
