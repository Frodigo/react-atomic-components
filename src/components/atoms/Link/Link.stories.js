import React from 'react';

import Link from './Link';

export default {
    title: 'Atoms/Buttons/Link',
    component: Link,
    argTypes: {
        variant: {
            control: {
                type: 'select',
                options: ['primary', 'secondary', 'text']
            }
        },
        active: {
            control: {
                type: 'select',
                options: [false, true]
            }
        }
    }
};

const Template = (args) => <Link {...args} />;

export const TextLink = Template.bind({});
TextLink.args = {
    variant: 'text',
    children: 'Text link',
    disabled: false,
    active: false,
    href: '#'
};

export const LinkAsButton = Template.bind({});
LinkAsButton.args = {
    variant: 'secondary',
    children: 'Link as button',
    disabled: false,
    active: false,
    href: '#'
};
