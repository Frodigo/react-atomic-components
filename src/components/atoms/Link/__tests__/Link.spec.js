import React from 'react';
import { render } from '@testing-library/react';

import Link from '../Link';

describe('Link should be rendered', () => {
    const text = 'text link';

    test('Link renders correctly', () => {
        const { container } = render(<Link href="#">{text}</Link>);

        expect(container).toMatchSnapshot();
    });
});
