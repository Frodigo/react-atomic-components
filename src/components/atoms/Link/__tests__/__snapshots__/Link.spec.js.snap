// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should be rendered Link renders correctly 1`] = `
<div>
  <a
    href="#"
  >
    text link
  </a>
</div>
`;
